python/paddle/v2/dataset/common.py

```diff
@@ -74,3 +76,59 @@ def fetch_all():
             getattr(
                 importlib.import_module("paddle.v2.dataset.%s" % module_name),
                 "fetch")()
+
+def split(reader, line_count, suffix="%05d.pickle"):
+    """
+    you can call the function as:
+
+    split(paddle.v2.dataset.cifar.train10(), line_count=1000,
+        suffix="imikolov-train-%05d.pickle")
+
+    the output files as:
+
+    |-imikolov-train-00000.pickle
+    |-imikolov-train-00001.pickle
+    |- ...
+    |-imikolov-train-00480.pickle
+
+    :param reader: the reader creator will be split
+    :param line_count: line count for each file
+    :param suffix: the suffix for each file,
+                    contain "%d" means the id for each file
+    """
+    lines = []
+    indx_f = 0
+    for i, d in enumerate(reader()):
+        lines.append(d)
+        if i >= line_count and i % line_count == 0:
+            with open(suffix % indx_f, "w") as f:
+                pickle.dump(lines, f)
+                lines = []
+                indx_f += 1
+    if not lines:
+        with open(suffix % indx_f, "w") as f:
+            pickle.dump(lines, f)
+
+def cluster_files_reader(files_pattern, trainers, trainer_id):
+    """
+    Create a reader that yield element from the given files, select
+    a file set according trainer count and trainer_id
+
+    :param files_pattern: the files which generating by split(...)
+    :param trainers: total trainer count
```

Comment:
I think `trianer_count` is a better name instead of `trainers` because I have to read the comment to understand the meaning of `trainers`.